pkg/reconciler/finalizer/finalizer.go

```diff
@@ -0,0 +1,605 @@
+	mapper meta.RESTMapper,
+	destroyer *applier.Applier,
+) *Controller {
+	return &Controller{
+		scope:         scope,
+		syncName:      syncName,
+		dynamicClient: dynamicClient,
+		mapper:        mapper,
+		destroyer:     destroyer,
+	}
+	// TODO: Update the reconciler-manager's finalizer to wait for the remediator's finalizer
+}
+
+// Start the Controller in a background goroutine.
+//
+// Inputs:
+//   ctx - a context, which will trigger shutdown of the Controller when
+//         cancelled.
+//   continueCh - a channel, which should be closed after the `stopCtx` is
+//                closed, and after the rest of the reconciler controllers (e.g.
+//                parser & remediator) have be stopped.
+//
+// Returns:
+//   stopCtx - a new context, which wraps the input context, and which will be
+//             cancelled when the deletionTimestamp is detected.
+//   doneCh - a new channel, which will be stopped when the Controller has
+//            stopped, either by cancellation of the input context or completion
+//            of finalization.
+//   err - any error encountered while starting the Controller
+func (c *Controller) Start(ctx context.Context, continueCh <-chan struct{}) (context.Context, <-chan struct{}, error) {
+	exampleObj := &unstructured.Unstructured{}
+	exampleObj.SetName(c.syncName)
+	if c.scope == declared.RootReconciler {
+		exampleObj.SetGroupVersionKind(kinds.RootSyncV1Beta1())
+		exampleObj.SetNamespace(configmanagement.ControllerNamespace)
+	} else {
+		exampleObj.SetGroupVersionKind(kinds.RepoSyncV1Beta1())
+		exampleObj.SetNamespace(string(c.scope))
+	}
+
+	mapping, err := c.mapper.RESTMapping(exampleObj.GroupVersionKind().GroupKind())
+	if err != nil {
+		return ctx, nil, errors.Wrapf(err, "REST mapping")
+	}
+
+	// Use a field selector to do client-side filtering by name,
+	// because the API doesn't support server-side filtering by "metadata.name".
+	fieldSelectorStr := fields.OneTermEqualSelector("metadata.name", exampleObj.GetName()).String()
+
+	// TODO: make configurable?
+	resyncPeriod := 1 * time.Hour
+
+	informer := cache.NewSharedIndexInformer(
+		&cache.ListWatch{
+			ListFunc: func(opts metav1.ListOptions) (runtime.Object, error) {
+				opts.FieldSelector = fieldSelectorStr
+				klog.V(3).Infof("Listing %s", objSummary(exampleObj))
+				uList, err := c.dynamicClient.Resource(mapping.Resource).
+					Namespace(exampleObj.GetNamespace()).
+					List(ctx, opts)
+				if err != nil {
+					return uList, err
+				}
+				return uList, nil
+			},
+			WatchFunc: func(opts metav1.ListOptions) (k8swatch.Interface, error) {
+				opts.Watch = true
+				opts.FieldSelector = fieldSelectorStr
+				klog.V(3).Infof("Watching %s", objSummary(exampleObj))
+				uWatch, err := c.dynamicClient.Resource(mapping.Resource).
+					Namespace(exampleObj.GetNamespace()).
+					Watch(ctx, opts)
+				if err != nil {
+					return uWatch, err
+				}
+				return uWatch, nil
+			},
+		},
+		exampleObj,
+		resyncPeriod,
```

Comment:
What is the meaning of `resyncPeriod` in this context?

Reply:
It's an informer implementation detail. When that period expires, it dumps the cache and does a new List to get all the latest objects. AFAIK, it's mostly obsolete, but I'm not sure if it's always safe to disable resync yet, or if it requires certain newer server-side changes.

Reply:
This is now obsolete. The controller-manager handles making and configuring he informers.